storage/src/test/java/org/apache/kafka/tiered/storage/TieredStorageTestContext.java

```diff
@@ -0,0 +1,322 @@
+        closeClients();
+        harness.killBroker(brokerId);
+        harness.startBroker(brokerId);
+        initContext();
+    }
+
+    public void stop(int brokerId) {
+        closeClients();
+        harness.killBroker(brokerId);
+        initContext();
+    }
+
+    public void start(int brokerId) {
+        closeClients();
+        harness.startBroker(brokerId);
+        initContext();
+    }
+
+    public void eraseBrokerStorage(int brokerId) throws IOException {
+        localStorages.get(brokerId).eraseStorage();
+    }
+
+    public TopicSpec topicSpec(String topicName) {
+        synchronized (topicSpecs) {
+            return topicSpecs.get(topicName);
+        }
+    }
+
+    public LocalTieredStorageSnapshot takeTieredStorageSnapshot() {
+        int aliveBrokerId = harness.aliveBrokers().head().config().brokerId();
+        return LocalTieredStorageSnapshot.takeSnapshot(remoteStorageManagers.get(aliveBrokerId));
+    }
+
+    public LocalTieredStorageHistory tieredStorageHistory(int brokerId) {
+        return remoteStorageManagers.get(brokerId).getHistory();
+    }
+
+    public List<LocalTieredStorage> remoteStorageManagers() {
+        return remoteStorageManagers;
+    }
+
+    public List<BrokerLocalStorage> localStorages() {
+        return localStorages;
+    }
+
+    public Deserializer<String> de() {
+        return de;
+    }
+
+    public Admin admin() {
+        return admin;
+    }
+
+    public boolean isActive(Integer brokerId) {
+        return harness.aliveBrokers().exists(b -> b.config().brokerId() == brokerId);
+    }
+
+    public boolean isAssignedReplica(TopicPartition topicPartition, Integer replicaId)
+            throws ExecutionException, InterruptedException {
+        String topic = topicPartition.topic();
+        int partition = topicPartition.partition();
+        TopicDescription description = admin.describeTopics(Collections.singletonList(topicPartition.topic()))
+                .allTopicNames().get().get(topic);
+        TopicPartitionInfo partitionInfo = description.partitions().get(partition);
+        return partitionInfo.replicas().stream().anyMatch(node -> node.id() == replicaId);
+    }
+
+    public Optional<UnifiedLog> log(Integer brokerId, TopicPartition partition) {
+        Option<UnifiedLog> log = harness.brokers().apply(brokerId).logManager().getLog(partition, false);
+        return log.isDefined() ? Optional.of(log.get()) : Optional.empty();
+    }
+
+    public void succeed(TieredStorageTestAction action) {
+        testReport.addSucceeded(action);
+    }
+
+    public void fail(TieredStorageTestAction action) {
+        testReport.addFailed(action);
+    }
+
+    public void printReport(PrintStream output) {
+        testReport.print(output);
+    }
+
+    @Override
+    public void close() throws IOException {
+        Utils.closeAll(producer, consumer);
+        Utils.closeQuietly(admin, "Admin client");
+    }
+
+    private void closeClients() {
```

Comment:
if we are creating clients using the harness, we don't have to close them explicitly. Harness takes care of that. It closes them at the end of the test.